docs/FeatureManagement/FeatureFlag.v2.0.0.schema.json

```diff
@@ -0,0 +1,345 @@
+          "type": "string",
+          "title": "Requirement Type",
+          "description": "Determines whether any or all registered client filters must be enabled for the feature to be considered enabled.",
+          "enum": [
+            "Any",
+            "All"
+          ],
+          "default": "Any"
+        },
+        "client_filters": {
+          "$id": "#/properties/conditions/properties/client_filters",
+          "type": "array",
+          "title": "Client Filter Collection",
+          "description": "Filters that must run on the client and be evaluated as true for the feature to be considered enabled.",
+          "items": {
+            "$id": "#/properties/conditions/properties/client_filters/items",
+            "type": "object",
+            "title": "Client Filter",
+            "required": [
+              "name"
+            ],
+            "properties": {
+              "name": {
+                "$id": "#/properties/conditions/properties/client_filters/items/properties/name",
+                "type": "string",
+                "title": "Client Filter Name",
+                "description": "The name used to refer to and require a client filter.",
+                "default": "",
+                "examples": [
+                  "Percentage",
+                  "TimeWindow"
+                ],
+                "pattern": "^(.*)$"
+              },
+              "parameters": {
+                "$id": "#/properties/conditions/properties/client_filters/items/properties/parameters",
+                "type": "object",
+                "title": "Client Filter Parameters",
+                "description": "Custom parameters for a given client filter. A client filter can require any set of parameters of any type.",
+                "required": [],
+                "patternProperties": {
+                  "^.*$": {
+                    "anyOf": [
+                      {
+                        "type": "string"
+                      },
+                      {
+                        "type": "null"
+                      },
+                      {
+                        "type": "object"
+                      },
+                      {
+                        "type": "number"
+                      },
+                      {
+                        "type": "array"
+                      },
+                      {
+                        "type": "boolean"
+                      }
+                    ]
+                  }
+                },
+                "additionalProperties": false
+              }
+            }
+          }
+        }
+      }
+    },
+    "variants": {
+      "$id": "#/properties/variants",
+      "type": "array",
+      "title": "Variants Collection",
+      "description": "The list of variants defined for this feature. A variant represents a value of a feature flag that can be a string, a number, a boolean, or even a configuration object.",
```

Comment:
A variant represents a **configuration** value of a feature flag that can be a string, a number, a boolean, or ~even~ a configuration object.